resteasy/src/test/scala/com/cerner/beadledom/resteasy/ModuleProcessorSpec.scala

```diff
@@ -0,0 +1,68 @@
+package com.cerner.beadledom.resteasy
+
+import org.jboss.resteasy.spi.{Registry, ResteasyProviderFactory}
+import org.scalatest.{FunSpec, MustMatchers}
+import org.scalatest.mockito.MockitoSugar
+import com.cerner.beadledom.guice.dynamicbindings.AnnotatedModule
+import com.google.inject.Guice
+
+class ModuleProcessorSpec extends FunSpec with MockitoSugar with MustMatchers{
+
+  val registry: Registry = mock[Registry]
+  val providerFactory: ResteasyProviderFactory = mock[ResteasyProviderFactory]
+
+  describe("ModuleProcessor#new") {
+    it("throws a NullPointerException for null registry") {
+      intercept[NullPointerException] {
+        new ModuleProcessor(null, providerFactory)
+      }
+    }
+
+    it("throws a NullPointerException for null provider factory") {
+      intercept[NullPointerException] {
+        new ModuleProcessor(registry, null)
+      }
+    }
+  }
+
+  describe("ModuleProcessor#processInjector") {
+    it("throws a NullPointerException for null injector") {
+      intercept[NullPointerException] {
+        new ModuleProcessor(registry, providerFactory).processInjector(null)
+      }
+    }
+
+    it("excludes providers that have a binding annotation") {
+      val injector = Guice.createInjector(new AnnotatedModule(classOf[TestBindingAnnotation]) {
+        override def configure(): Unit = {
+          bind(classOf[TestJaxRsProvider]).annotatedWith(getBindingAnnotation).toProvider(
+            new TestProviderWithBindingAnnotation(getBindingAnnotation)
+          )
+          expose(classOf[TestJaxRsProvider]).annotatedWith(getBindingAnnotation)
+        }
+      })
+      val providerFactory = new ResteasyProviderFactory()
+      new ModuleProcessor(registry, providerFactory).processInjector(injector)
+
+      providerFactory.getProviderInstances mustBe empty
+    }
+
+    it("excludes providers that have a binding annotation and registers providers without a binding annotation") {
+      val injector = Guice.createInjector(new AnnotatedModule(classOf[TestBindingAnnotation]) {
+        override def configure(): Unit = {
+          bind(classOf[TestJaxRsProvider]).annotatedWith(getBindingAnnotation).toProvider(
+            new TestProviderWithBindingAnnotation(getBindingAnnotation)
+          )
+          expose(classOf[TestJaxRsProvider]).annotatedWith(getBindingAnnotation)
+
+          bind(classOf[TestJaxRsProvider]).toProvider(new TestProvider())
+          expose(classOf[TestJaxRsProvider])
+        }
+      })
+      val providerFactory = new ResteasyProviderFactory()
+      new ModuleProcessor(registry, providerFactory).processInjector(injector)
+
+      providerFactory.getProviderInstances must have size 1
```

Comment:
Would it be possible to use 2 different JAX-RS provider classes in the Guice module above, and then have an assertion on the particular class/instance that is expected here?

Reply:
https://github.com/cerner/beadledom/pull/143/commits/75d364f068be5b48f12a864aa2a8bd938fb929b8